test/ziggurat/messaging/consumer_test.clj

```diff
@@ -248,51 +237,70 @@
           (with-open [ch (lch/open connection)]
             (let [queue-name          (get-in (rabbitmq-config) [:dead-letter :queue-name])
                   prefixed-queue-name (str topic-entity-name "_" queue-name)
                   [meta payload]      (lb/get ch prefixed-queue-name false)
-                  _                   (process-message-from-queue ch meta payload topic-entity processing-fn)
+                  _                   (consumer/process-message-from-queue ch meta payload topic-entity processing-fn)
                   consumed-message    (util/get-msg-from-dead-queue-without-ack topic-entity-name)]
               (is (= false @processing-fn-called))
               (is (= consumed-message nil))))))))
   (testing "process-message function should reject and re-queue a message if processing fails. It should also report the error"
     (fix/with-queues {topic-entity {:handler-fn #(constantly nil)}}
-      (let [message       (gen-message-payload topic-entity)
-            processing-fn (fn [message-arg]
-                            (is (= message-arg message))
-                            (throw (Exception. "exception message")))
+      (let [message           (gen-message-payload topic-entity)
+            processing-fn     (fn [message-arg]
+                                (is (= message-arg message))
+                                (throw (Exception. "exception message")))
             topic-entity-name (name topic-entity)
-            report-fn-called?  (atom false)]
+            report-fn-called? (atom false)]
         (with-redefs [report-error (fn [_ _] (reset! report-fn-called? true))]
           (producer/publish-to-dead-queue message)
           (with-open [ch (lch/open connection)]
             (let [queue-name          (get-in (rabbitmq-config) [:dead-letter :queue-name])
                   prefixed-queue-name (str topic-entity-name "_" queue-name)
                   [meta payload]      (lb/get ch prefixed-queue-name false)
-                  _                   (process-message-from-queue ch meta payload topic-entity processing-fn)
+                  _                   (consumer/process-message-from-queue ch meta payload topic-entity processing-fn)
                   consumed-message    (util/get-msg-from-dead-queue-without-ack topic-entity-name)]
               (is (= consumed-message message))
               (is @report-fn-called?))))))))
+
+(deftest convert-and-ack-message-test
+  (testing "While constructing a MessagePayload, adds topic-entity as a keyword and retry-count as 0 if message does not already has :retry-count"
+    (let [message                   {:foo "bar"}
+          converted-message-payload (consumer/convert-and-ack-message nil {:delivery-tag "delivery-tag"} (nippy/freeze message) false "default")]
+      (is (= converted-message-payload message))))
+  (testing "While constructing a MessagePayload, adds topic-entity as a keyword and retry-count as it exists in the message"
```

Comment:
we don't need this test. The code for this has been removed.

Reply:
Same as above

Reply:
done